### DISCUSSION TITLE
Python FastAPI deployment with Vercel

### DISCUSSION THREAD

**Discussion by laurentiupiciu** [vercel/community#4692] [Help] [answered]
### Summary

Hi everyone
Has anybody experience with deploying python apps on vercel?

I see that the only possibility is for a serverless function. My question is how can one increase the `maxDuration`? Now, the default is 15 and I do not find any resource that explains how that can be done, hence I think it's not possible. But I just want to be sure of that.

My vercel.json is as following:
```
{
    "devCommand": "uvicorn server:app --host 0.0.0.0 --port 3000",
    "builds": [
        {
            "src": "server.py",
            "use": "@vercel/python"
        }
    ],
    "routes": [
        {
            "src": "/(.*)",
            "dest": "server.py"
        }
    ]
}
```

### Steps to Reproduce

-

**Comment by amyegan** [comment, marked as answer]
Hi @laurentiupiciu! You can find info about setting `maxDuration` in our Runtimes documentation.

https://vercel.com/docs/functions/serverless-functions/runtimes#max-duration

**Reply by giux78** [reply]
Hi I have the same problem, I use the vercel.json below but the maxDurations configuration is not taken in the deployment. Is only supported in the new vercel.json style using function or the is also a way using the legacy vercel.json. In any case in my python server is not possible to use the new vercel.json syntax. Could you give us an answer I have seen more than one related question on github but not a clear answer if it is possible. Thanks in advance.

{
  "builds": [
    {
      "src": "app.py",
      "use": "@vercel/python",
      "config": { "maxDurations": 300 }
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "app.py"
    }
  ]
}

**Reply by u8621011** [reply]
I finally found the solution, please check the official vercel.json settings sample here :

[https://github.com/vercel-support/vercel-fastapi/blob/main/vercel.json](https://github.com/vercel-support/vercel-fastapi/blob/main/vercel.json)
